addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1064 @@
+            # Localized day name
+            res = babel.dates.format_date(date=date, format='EEEE')
+
+        elif interval == 'time':
+            # Localized time
+
+            dummy, format_time = self.get_date_formats()
+            res = tools.ustr(date.strftime(format_time + " %Z"))
+
+        return res
+
+    @api.model
+    def search(self, args, offset=0, limit=0, order=None, count=False):
+
+        if self.env.context.get('mymeetings'):
+            partner_id = self.env.user.partner_id.id
+            args += [('partner_ids', 'in', [partner_id])]
+
+        new_args = []
+        for arg in args:
+            new_arg = arg
+
+            if arg[0] in ('start_date', 'start_datetime', 'start',) and arg[1] == ">=":
+                if self.env.context.get('virtual_id', True):
+                    new_args += ['|', '&', ('recurrency', '=', 1), ('final_date', arg[1], arg[2])]
+            elif arg[0] == "id":
+                new_id = get_real_ids(arg[2])
+                new_arg = (arg[0], arg[1], new_id)
+            new_args.append(new_arg)
+
+        if not self.env.context.get('virtual_id', True):
+            return super(CalendarEvent, self).search(new_args, offset=offset, limit=limit, order=order, count=count)
+        # offset, limit, order and count must be treated separately as we may need to deal with virtual ids
+        res = super(CalendarEvent, self).search(new_args, offset=0, limit=0, order=None, count=False)
+        res = res.get_recurrent_ids(args, order=order)
+        if count:
+            return len(res)
+        elif limit:
+            return res[offset: offset + limit]
+        return res
+
+    @api.multi
+    def copy(self, default=None):
+        default = dict(default or {})
+        self._set_date(default, id=default.get('id'))
+        return super(CalendarEvent, self).copy(default)
+
+    @api.multi
+    def _detach_one_event(self, values=dict()):
+        real_event_id = calendar_id2real_id(self.id)
+        data = self.read(['allday', 'start', 'stop', 'rrule', 'duration'])[0]
+        data['start_date' if data['allday'] else 'start_datetime'] = data['start']
+        data['stop_date' if data['allday'] else 'stop_datetime'] = data['stop']
+        if data.get('rrule'):
+            data.update(
+                values,
+                recurrent_id=real_event_id,
+                recurrent_id_date=data.get('start'),
+                rrule_type=False,
+                rrule='',
+                recurrency=False,
+                final_date=fields.Datetime.from_string(data.get('start')) + timedelta(hours=values.get('duration', False) or data.get('duration'))
+            )
+
+            #do not copy the id
+            if data.get('id'):
+                del(data['id'])
+            new_id = self.browse(real_event_id).copy(default=data)
+            return new_id
+
+    @api.multi
+    def open_after_detach_event(self):
+        new_id = self._detach_one_event(self.ids[0])
```

Comment:
Its method called on button, we can set ensure_one here to call it for singleton and then you can avoid ids[0]..